internal/email/email.service.go

```diff
@@ -0,0 +1,211 @@
+package email
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"net/smtp"
+	"time"
+
+	"github.com/brain-flowing-company/pprp-backend/apperror"
+	"github.com/brain-flowing-company/pprp-backend/config"
+	"github.com/brain-flowing-company/pprp-backend/internal/enums"
+	"github.com/brain-flowing-company/pprp-backend/internal/models"
+	"github.com/brain-flowing-company/pprp-backend/utils"
+	"go.uber.org/zap"
+)
+
+type Service interface {
+	SendVerificationEmail(string) *apperror.AppError
+	VerifyEmail(*models.EmailVerificationRequest) (string, *apperror.AppError)
+}
+
+type serviceImpl struct {
+	repo   Repository
+	logger *zap.Logger
+	cfg    *config.Config
+}
+
+func NewService(logger *zap.Logger, cfg *config.Config, repo Repository) Service {
+	return &serviceImpl{
+		repo,
+		logger,
+		cfg,
+	}
+}
+
+func (s *serviceImpl) SendVerificationEmail(userEmail string) *apperror.AppError {
+
+	if !utils.IsValidEmail(userEmail) {
+		return apperror.
+			New(apperror.InvalidEmail).
+			Describe("Invalid email")
+	}
+
+	var countEmail int64
+	findEmailErr := s.repo.CountEmail(&countEmail, userEmail)
+	if findEmailErr != nil {
+		s.logger.Error("Could not count email", zap.Error(findEmailErr))
+		return apperror.
+			New(apperror.InternalServerError).
+			Describe("Could not send email. Please try again later")
+	} else if countEmail > 0 {
+		return apperror.
+			New(apperror.EmailAlreadyExists).
+			Describe("Email already exists")
+	}
+
+	code := "SCK-" + utils.RandomString(16)
+
+	expiredAt := time.Now().Add(5 * time.Minute)
```

Comment:
Use config variable over embedding into code, so we can change verification lifetime later in env var without re-compiling our code.